File: python/3442.py
# Example 1:

# Input: s = "aaaaabbc"

# Output: 3

# Explanation:

# The character 'a' has an odd frequency of 5, and 'b' has an even frequency of 2.
# The maximum difference is 5 - 2 = 3.
# Example 2:

# Input: s = "abcabcab"

# Output: 1

# Explanation:

# The character 'a' has an odd frequency of 3, and 'c' has an even frequency of 2.
# The maximum difference is 3 - 2 = 1.
 

# Constraints:

# 3 <= s.length <= 100
# s consists only of lowercase English letters.
# s contains at least one character with an odd frequency and one with an even frequency.

class Solution(object):
    def maxDifference(self, s):
        """
        :type s: str
        :rtype: int
        """
        if not (3 <= len(s) <= 100):
            return "s must contain at least 3 and at most 100 characters"
        if not s.islower():
            return "s all character must be in lowercase"
        
                # Manually count character frequencies
        freq = {}
        for char in s:
            if char in freq:
                freq[char] += 1
            else:
                freq[char] = 1
        
        # print(freq)
        
        # Extract odd and even frequency values
        odd_freq = [v for v in freq.values() if v % 2 == 1]
        even_freq = [v for v in freq.values() if v % 2 == 0]

        # Ensure at least one odd and one even frequency exists
        if odd_freq and even_freq:
            return max(odd_freq) - min(even_freq)
        else:
            return "Invalid input: need at least one odd and one even frequency"
sol = Solution()
# s = "abcabcab"
s = "aaaaabbc"
print(sol.maxDifference(s))

File: C/3442.c
#include <stdio.h>
#include <string.h>

#define MAX_CHAR 26

int maxDifference(char *s) {
    int n = strlen(s);
    
    // Constraint check
    if (n < 3 || n > 100) {
        printf("s must contain at least 3 and at most 100 characters\n");
        return -1;
    }

    // Frequency array to count occurrences of each character
    int freq[MAX_CHAR] = {0};
    
    // Count frequencies
    for (int i = 0; i < n; i++) {
        if (s[i] < 'a' || s[i] > 'z') {
            printf("s all characters must be in lowercase\n");
            return -1;
        }
        freq[s[i] - 'a']++;
    }

    // Find max odd and min even frequencies
    int max_odd = -1, min_even = 101;

    for (int i = 0; i < MAX_CHAR; i++) {
        if (freq[i] > 0) {
            if (freq[i] % 2 == 1) {
                if (freq[i] > max_odd) {
                    max_odd = freq[i];
                }
            } else {
                if (freq[i] < min_even) {
                    min_even = freq[i];
                }
            }
        }
    }

    // Ensure at least one odd and one even frequency exist
    if (max_odd != -1 && min_even != 101) {
        return max_odd - min_even;
    } else {
        printf("Invalid input: need at least one odd and one even frequency\n");
        return -1;
    }
}

int main() {
    char s[] = "aaaaabbc";  // Example input
    int result = maxDifference(s);
    if (result != -1) {
        printf("Output: %d\n", result);
    }
    return 0;
}
